### DISCUSSION TITLE
Human AI is bad at fighting indoor intruders

### DISCUSSION THREAD

**Discussion by A2ur3** [FakeFishGames/Barotrauma#15073] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

https://github.com/user-attachments/assets/0af52f38-157f-48f5-b599-4290c474c165
issue 1: security officer uses diving knife to attack enemy although his revolver still has rounds (AI should try to retreat to a appropriate distance to use ranged weapon. combat priority of revolver is higher than diving knife)
issue 2: HR manager uses wrench as a weapon to attack enemy which prevents security officer behind her from firing

### Reproduction steps

1. go to outpost
2. spawn Husk_exosuit and watch

### Bug prevalence

Just once

### Single player or multiplayer?

Single player

### Version

v1.6.19.1 (Unto the Breach Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by SomeRandomNoobKekeke** [comment]
Also this https://github.com/FakeFishGames/Barotrauma/discussions/13903
![BaroDev (ultra wide)](https://github.com/user-attachments/assets/e6b5a8f8-7f7f-44fc-b222-a2eeaf6c7d58)

**Comment by pavel4win** [comment]
And this
https://github.com/FakeFishGames/Barotrauma/discussions/14914

**Comment by pavel4win** [comment]
I noticed that bots are good at dealing with raptors, they immediately start shooting at them with SMG and a revolver.
I noticed that bots rush at crawlers and tigers with a screwdriver, even if they have a loaded SMG in their inventory.
I'll share another masterpiece of artificial intelligence.
Today, a tiger got inside a boat.
The bot rushed at it with a screwdriver, while it had a loaded SMG.
When I started shooting at this monster in the hopes of killing the tiger before it killed the bot,
I accidentally hit the bot that was poking the monster with a screwdriver.
Then the bot started shooting at me with an SMG and killed me.
Just incredible AI.

**Comment by itchyOwl** [comment]
It's intentional that bots use melee weapons instead of ranged weapons when they are so close to the enemy that they can't hit it with ranged weapons. I agree that an ideal behavior in some cases would be to retreat a bit so that they could shoot at it. And actually this is how bots that don't have an active fight intruders objective should behave. But I'm not sure it would work in all circumstances when the bot has an offensive objective, i.e. they are expected to actively attack the enemies. Currently there's no logic for retreating while still being in the offensive. Might give it a test at some point.

Some outpost NPCs, such as a HR manager, have unnecessary [NPC Wait Actions that make them unresponsive to threats](https://github.com/FakeFishGames/Barotrauma/discussions/13960), which I think is causing them to just stand there, regardless of the monster. Should be fixed in the unstable soon. So, I'd say this shouldn't happen with bots.

> I noticed that bots are good at dealing with raptors, they immediately start shooting at them with SMG and a revolver.
I noticed that bots rush at crawlers and tigers with a screwdriver, even if they have a loaded SMG in their inventory.

It shouldn't matter which monster they are facing. The behavior shouldn't be any different. So I'd say conclusions are incorrect. Without further information, it's difficult to say what's happening there, but I think they probably just try to pass 

> Then the bot started shooting at me with an SMG and killed me.

Bots shouldn't ever intentionally shoot at you. In mp, if there's no enemies around, and a player attacks a bot, they might try to arrest the player. In single player, they will just run away. So, I'm pretty sure these shots were coincidental: the bot attempted to shoot at the monster.

**Reply by pavel4win** [reply]
Maybe it's some kind of bug in the code, but the bot shot at me after I attacked it by accident. The monster was already dead at that time. This happened twice in the network game in different rounds. I'm sure of it.

It's strange that bots don't distinguish which monster is attacking them, a tiger is much more dangerous than a crawler.
A bot shouldn't attack a tiger with a screwdriver at all, it should run away while it has the opportunity, if it doesn't have a firearm.
it would be reasonable to attack the tiger with the most powerful weapon and if the bot has an SMG why does it use a screwdriver at all? After all, you can shoot point-blank?

**Comment by itchyOwl** [comment]
Ticketed.

**Comment by NilanthAnimosus** [comment]
Closing as fixed and merged (coming in the next update and should already be in unstable).

They generally are much better at switching weapons back to ranged from melee now, and have improvements in group combat (not always, but sometimes now they will crouch and such to allow those from behind to attack, and close the distance if someone blocks a shot at an angle at times, amongst other small improvements).
